### PR TITLE
[FSSDK-9973] doc: update readme with webhook & notification synchronization doc

### PR DESCRIPTION
**Pull request by pulak-opti**
## Summary
- updated readme with webhook & notification synchronization doc

N.B. The official documentation will be updated with the same content once the PR is approved.

## Ticket
- [FSSDK-9973](https://jira.sso.episerver.net/browse/FSSDK-9973)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +186,26 @@ To accept webhook requests Agent must be configured by mapping an Optimizely Pro
 with the associated secret used for validating the inbound request. An example webhook configuration can
 be found in the the provided [config.yaml](./config.yaml).
 
+When running Agent in High Availability (HA) mode, it's important to ensure that all nodes are updated promptly when a webhook notification is received. By default, only one Agent node or instance will receive the webhook notification. A pub-sub system can be used to ensure this.
+Redis, a powerful in-memory data structure store, can be used as a relay to propagate the webhook event to all other nodes in the HA setup. This ensures that all nodes are notified about the event and can update their data files accordingly.
+To set up Redis as a relay, you need to enable the data file synchronization in your Optimizely Agent configuration. The PubSub feature of Redis is used to publish the webhook notifications to all subscribed Agent nodes.
+Here's an example of how you can enable the datafile synchronization with Redis:
```
Comment (mikechu-optimizely) [inline README.md:192]: NIT: do you want to separate each of these lines to give some whitespace breathing room to the excellent instructions?

**Review by mikechu-optimizely — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (pulak-opti) [reply README.md]: updated, thanks @mikechu-optimizely :raised_hands:

File: README.md [hunk 2]
```diff
@@ -186,6 +186,29 @@ To accept webhook requests Agent must be configured by mapping an Optimizely Pro
 with the associated secret used for validating the inbound request. An example webhook configuration can
 be found in the the provided [config.yaml](./config.yaml).
 
+When running Agent in High Availability (HA) mode, it's important to ensure that all nodes are updated promptly when a webhook notification is received. By default, only one Agent node or instance will receive the webhook notification. A pub-sub system can be used to ensure this.
```
Comment (jaeopt) [inline README.md:189]:
```suggestion
When running Agent in High Availability (HA) mode, it's important to ensure that all nodes are updated promptly when a webhook event (datafile updated) is received. By default, only one Agent node or instance will receive the webhook notification. A pub-sub system can be used to ensure this.
```

File: README.md [hunk 3]
```diff
@@ -186,6 +186,29 @@ To accept webhook requests Agent must be configured by mapping an Optimizely Pro
...
+
+Redis, a powerful in-memory data structure store, can be used as a relay to propagate the webhook event to all other nodes in the HA setup. This ensures that all nodes are notified about the event and can update their data files accordingly.
```
Comment (jaeopt) [inline README.md:191]:
```suggestion
Redis, a powerful in-memory data structure store, can be used as a relay to propagate the datafile webhook event to all other nodes in the HA setup. This ensures that all nodes are notified about the event and can update their datafiles accordingly.
```

Comment (jaeopt) [reply README.md:191]: I know this is generally true for all types of webhook, but "datafile webhook" may be helpful for following context.

File: README.md [hunk 4]
```diff
@@ -186,6 +186,29 @@ To accept webhook requests Agent must be configured by mapping an Optimizely Pro
...
+
+To set up Redis as a relay, you need to enable the data file synchronization in your Optimizely Agent configuration. The PubSub feature of Redis is used to publish the webhook notifications to all subscribed Agent nodes.
```
Comment (jaeopt) [inline README.md:193]:
```suggestion
To set up Redis as a relay, you need to enable the datafile synchronization in your Optimizely Agent configuration. The PubSub feature of Redis is used to publish the webhook notifications to all subscribed Agent nodes.
```

File: README.md (hunk 2 above)
Comment (jaeopt) [reply README.md:189]: "notification" can be confused with other notification sync below.

**Review by jaeopt — CHANGES_REQUESTED** [review]
A few small changes suggested

**Review by jaeopt — APPROVED** [review]
LGTM

**Pull request merged**